source/fundamentals/data-formats/document-data-format-bson.txt

```diff
@@ -0,0 +1,44 @@
+==========================
+Document Data Format: BSON
+==========================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+
+
+Overview
+--------
+
+- Definition of BSON format, where it's used, why it's used.
+- Link to more info on BSON
+- What information you can find in the following sections
+
+Installation
+------------
+
+- Instructions on how to add the dependency with Maven or Gradle
+
+BsonReader
```

Comment:
I don't remember anymore: did we decide to lead with BsonReader and BsonWriter?  I thought we decided that it was a necessary pre-cursor to understanding how to write a custom Codec implementation, but otherwise it's not widely used.

Reply:
It's here: https://docs.google.com/document/d/1yUZenj7N-O8-R3TB_zqvuKljoEQS_T27Sww61-0mWqs/edit?disco=AAAAG5e8kTc
From reviewing this documentation, I thought it would be reasonable to include the same level of detail as this page without discussing Codecs: https://mongodb.github.io/mongo-java-driver/4.1/bson/readers-and-writers/

If it's not widely used, then it sounds like it would be better in the Codecs section. Let me know which you prefer.

Reply:
There's a resolved comment in our brainstorming doc that says "Might be better to include with Codec encode/decode".   I don't recall why it was resolved.  

It's only used by Codec implementations, so I think it's better to put it later.

Reply:
Thanks, moved to Custom Codec Example in codecs.txt